src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
Yeah, I think we could just make `:JRubyMetricsService` a required dependency to avoid the need to check for `nil`.  I'd been holding onto the thought, though, that it might still be useful -- I guess just if memory usage / perf to collect measurements were too excessive for some users? -- to have a way to disable JRuby metrics functionality by removing the `jruby-metrics-service` from the bootstrap config.  But maybe we're far enough down the road on this now with evidence of this not having needed to be done for any PE users that it's not worth preserving that capability.